libcaf_core/caf/ipv6_endpoint.hpp

```diff
@@ -18,19 +18,26 @@
 
 #pragma once
 
+#include <cstdint>
+#include <functional>
+
 #include "caf/detail/comparable.hpp"
+#include "caf/fwd.hpp"
 #include "caf/ipv6_address.hpp"
 #include "caf/meta/type_name.hpp"
 
 namespace caf {
 
 /// An IP endpoint that contains an ::ipv6_address and a port.
-class ipv6_endpoint : detail::comparable<ipv6_endpoint> {
+class ipv6_endpoint : detail::comparable<ipv6_endpoint>,
+                      detail::comparable<ipv6_endpoint, ipv4_endpoint> {
```

Comment:
Wouldn't it be useful to add this to the ipv4_endpoint too? Now one would have to compare `ipv6_endpoint`s with `ipv4_endpoint`s like this: `ipv6 > ipv4` but comparing like `ipv4 < ipv6` would not be possible, wouldn't it?

Reply:
That would actually produce ambiguous overloads. `comparable` injects operators for both directions.

Reply:
Nevermind. I just figured this out. 👍